swarm.go

```diff
@@ -22,9 +22,16 @@ type SwarmConfig struct {
 	// EnableAutoRelay enables the "auto relay user" feature.
 	// Node will find and use advertised public relays when it determines that
 	// it's not reachable from the public internet.
+	//
+	// Deprecated: This flag is deprecated and is overriden by
+	// `Swarm.AutoRelay.Enabled` if specified.
 	EnableAutoRelay bool
```

Comment:
nit: we should do what we already did for `DisableRelay` and hide it from new configs:
```suggestion
	EnableAutoRelay bool `json:",omitempty"`
```